packages/fuels-programs/src/logs.rs

```diff
@@ -87,14 +69,75 @@ impl LogDecoder {
     }
 }
 
+trait FilterMapLogReceipts {
+    type Output: Iterator<Item = ((Bech32ContractId, u64), Vec<u8>)>;
+    fn filter_map_log_receipts(self) -> Self::Output;
```

Comment:
As mentioned above, we  use `DecodableLog` elsewhere. We could use the same name here but I would avoid it because of possible confusion.

How about something like this:
```rust
trait ExtractLogIdData {
    type Output: Iterator<Item = (LogId, Vec<u8>)>;
    fn extract_log_id_with_data(self) -> Self::Output;
}
```